libvast/include/vast/command.hpp

```diff
@@ -192,8 +192,7 @@ struct invocation {
 /// @returns an error for malformed input, `none` otherwise.
 /// @relates command
 caf::expected<invocation>
-parse(const command& root, command::argument_iterator first,
-      command::argument_iterator last);
+parse(const command& root, std::vector<std::string>& arguments);
```

Comment:
I think `arguments` should really be taken by value. Alternatively you could also revert to using the iterators and just forcing the copy. This code is basically as far away from performance critical as it gets.